File: sdk/typescript/src/types/rerank.ts
import type { AgentCCMetadata } from "./shared.js";

export interface RerankResult {
  index: number;
  relevance_score: number;
  document?: { text: string };
}

export interface RerankResponse {
  id?: string;
  results: RerankResult[];
  model: string;
  usage?: { total_tokens: number };
  agentcc?: AgentCCMetadata;
  [key: string]: unknown;
}

export interface RerankParams {
  model: string;
  query: string;
  documents: string[] | { text: string }[];
  top_n?: number;
  return_documents?: boolean;
  extra_headers?: Record<string, string>;
  [key: string]: unknown;
}

File: sdk/python/src/agentcc/types/rerank.py
"""Rerank response types."""

from __future__ import annotations

from pydantic import BaseModel, ConfigDict


class RerankResult(BaseModel):
    model_config = ConfigDict(extra="allow")

    index: int
    relevance_score: float
    document: str | None = None  # Only present if return_documents=True


class RerankResponse(BaseModel):
    model_config = ConfigDict(extra="allow")

    id: str | None = None
    results: list[RerankResult] = []
    model: str | None = None
    usage: dict | None = None  # Some providers return token usage
